app/src/App.tsx

```diff
@@ -56,6 +57,28 @@ function App() {
   // An error message to display to the user.
   const [drawerOpen, setDrawerOpen] = useState(false);
 
+  useEffect(() => {
+    function setLocalStorageFromExternalDomain(message: MessageEvent) {
+      if (message.origin !== APPROVED_EXTERNAL_DOMAIN) {
+        return;
+      }
```

Comment:
It would be more secure to handle this on the server side in `cors.rs`. But I think an even better way would be to do what rust playground does:
- upload the code to a gist
- use the gist ID as a query parameter
- send the toolchain name as a query parameter as well
- sway playground fetches the code from the gist

Example:
`https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=3c65be25f3334da7947b0e120dce23a1`

For us it can be a little simpler: 
`https://sway-playground.org/?toolchain=latest&gist=3c65be25f3334da7947b0e120dce23a1`

Sway playground will need to sanitize the string. But then we don't have to worry about cors, and there's no concerns about timing issues.

I would recommend using React router dom's [useSearchParams](https://reactrouter.com/en/main/hooks/use-search-params) for reading and writing the query params.

Reply:
yeah this definitely seems like a better approach thanks for the feedback!